src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -373,63 +433,60 @@ private ReferenceGenerator GetReferenceGenerator(ISymbol typeParameterContext)
                     return _referenceGenerator;
                 }
 
+                /// <summary>
+                /// If we hit anything we don't know about, indicate failure.
+                /// </summary>
                 public override bool DefaultVisit(ISymbol symbol)
-                {
-                    throw new InvalidOperationException("Cannot generated a documentation comment id for symbol.");
-                }
+                    => false;
```

Comment:
Crux of the change.  no throwing exceptions for expected cases.  instead, we just indicate we cannot proceed.